app/webpacker/scripts/autocomplete_sort.spec.js

```diff
@@ -0,0 +1,62 @@
+import sort from '../../components/form_components/autocomplete/sort.js'
+
+describe('sort', () => {
+  it('returns an empty array for no matches', () => {
+    const options = [{ name: 'masters art' }]
+    expect(sort('degree', options)).toEqual([])
+  })
+
+  it('sorts alphabetically if weights are equal', () => {
+    const options = [
+      { name: 'masters engineering' }
+      { name: 'masters art' },
+    ]
+    expect(sort('masters', options)).toEqual(['masters art', 'masters engineering'])
+  })
+
+  it('prioritises exact name matches over synyonm matches', () => {
+    const options = [
+      { name: 'engineering' },
+      { name: 'art', synonyms: ['engineering'] }
+    ]
+    expect(sort('engineering', options)).toEqual(['engineering', 'art'])
+  })
+
+  it('prioritises exact synonym matches over name starts with', () => {
+    const options = [
+      { name: 'masters engineering' },
+      { name: 'art', synonyms: ['masters'] }
+    ]
+
+    expect(sort('masters', options)).toEqual(['art', 'masters engineering'])
+  })
+
+  it('prioritises name starts with over synonym starts with', () => {
+    const options = [
+      { name: 'masters engineering' },
+      { name: 'art', synonyms: ['masters engineering'] }
+    ]
+    expect(sort('masters', options)).toEqual(['masters engineering', 'art'])
+  })
+
+  it('prioritises synonym starts with over word in name starts with', () => {
+    const options = [
+      { name: 'masters engineering' },
+      { name: 'art', synonyms: ['engineering'] }
+    ]
+    expect(sort('engin', options)).toEqual(['art', 'masters engineering'])
+  })
+
+  it('requires all parts of the query to be matched', () => {
+    const options = [{ name: 'masters', synonyms: [] }]
+    expect(sort('masters art', options)).toEqual([])
+  })
+
+  it('allows custom boosting of results', () => {
```

Comment:
Its a bit confusing trying to figure out how much you need to boost something by in order to get the desired behaviour. Maybe we could make this a boolean instead? Struggling to think of situations where we would want different levels of boosting for different options.

Reply:
I think @edwardhorsford wants to be able to do just that - be able to tweak it in certain (as yet unknown) situations.

Reply:
The general idea would be that a team could 'tune' this to whatever their usage is. There's no right or wrong amount - it's very subjective.

When I originally did a similar algorithm some years ago I chose the various default scores so that a 'double' boost would make a lower quality score bump up quite a bit, but possibly still end up with a lower score than an exact match.

Reply:
I can add a code comment saying something similar to make it clear it is subjective.

Reply:
Comment added.